R/gather_fails.R

```diff
@@ -0,0 +1,169 @@
+#'
+#' @examples
+#' library(shiny)
+#' library(shinyvalidate)
+#'
+#' ui <- fluidPage(
+#'   selectInput("method", "validation method", c("hierarchical", "combined", "grouped")),
+#'   sidebarLayout(
+#'     sidebarPanel(
+#'       selectInput("letter", "select a letter:", c(letters[1:3], LETTERS[4:6])),
+#'       selectInput("number", "select a number:", 1:6),
+#'       br(),
+#'       selectInput("color", "select a color:",
+#'                   c("black", "indianred2", "springgreen2", "cornflowerblue"),
+#'                   multiple = TRUE),
+#'       sliderInput("size", "select point size:",
+#'                   min = 0.1, max = 4, value = 0.25)
+#'     ),
+#'     mainPanel(plotOutput('plot'))
+#'   )
+#' )
+#'
+#' server <- function(input, output) {
+#'   # set up input validation
+#'   iv <- InputValidator$new()
+#'   iv$add_rule("letter", sv_in_set(LETTERS, "choose a capital letter"))
+#'   iv$add_rule("number", ~ if (as.integer(.) %% 2L == 1L) "choose an even number")
+#'   iv$enable()
+#'   # more input validation
+#'   iv_par <- InputValidator$new()
+#'   iv_par$add_rule("color", sv_required(message = "choose a color"))
+#'   iv_par$add_rule("color", ~ if (length(.) > 1L) "choose only one color")
+#'   iv_par$add_rule("size", sv_between(left = 0.5, right = 3,
+#'                                      message_fmt = "choose a value between {left} and {right}"))
+#'   iv_par$enable()
+#'
+#'
+#'   output$plot <- renderPlot({
+#'     # validate output
+#'     switch(input[["method"]],
+#'            "hierarchical" = {
+#'              gather_fails(iv)
+#'              gather_fails(iv_par, "Set proper graphical parameters")
+#'            },
+#'            "combined" = gather_fails_com(iv, iv_par),
+#'            "grouped" = gather_fails_grp(list(
+#'              "Some inputs require attention" = iv,
+#'              "Set proper graphical parameters" = iv_par
+#'            )))
+#'
+#'     plot(eruptions ~ waiting, faithful, las = 1, pch = 16,
+#'          col = input[["color"]], cex = input[["size"]])
+#'   })
+#' }
+#'
+#' if (interactive()) {
+#'   shinyApp(ui, server)
+#' }
+
+
+#' @rdname gather_fials
+#' @export
+gather_fails <- function(iv, header = "Some inputs require attention", ...) {
+  checkmate::assert_class(iv, "InputValidator")
+  checkmate::assert_string(header, null.ok = TRUE)
+
+  fail_messages <- gather_messages(iv)
+  failings <- add_header(fail_messages, header)
+
+  shiny::validate(shiny::need(is.null(failings), failings), ...)
+}
+
+
+#' @rdname gather_fials
+#' @export
+gather_fails_com <- function(..., header = "Some inputs require attention") {
+  vals <- list(...)
+  lapply(vals, checkmate::assert_class, "InputValidator")
+  checkmate::assert_string(header, null.ok = TRUE)
+
+  fail_messages <- unlist(lapply(vals, gather_messages))
+  failings <- add_header(fail_messages, header)
+
+  shiny::validate(shiny::need(is.null(failings), failings))
+}
+
+
+#' @rdname gather_fials
```

Comment:
```suggestion
#' @rdname gather_fails
```

Reply:
Thanks for catching these, probably the last place I would look for typos 🙄